### PR TITLE
TASK-807

### PR DESCRIPTION
**Pull request by pfurio**
TASK-807

### CODE REVIEW

**Comment by pfurio** [comment]
Task linked: [TASK-807 JWT secret key issue](https://app.clickup.com/t/36631768/TASK-807)

File: opencga-catalog/src/main/java/org/opencb/opencga/catalog/db/mongodb/MetaMongoDBAdaptor.java [hunk 1]
```diff
@@ -221,20 +230,14 @@ public OpenCGAResult updateJWTParameters(ObjectMap params) throws CatalogDBExcep
 
         Document adminDocument = new Document();
         if (StringUtils.isNotEmpty(params.getString(SECRET_KEY))) {
+            // Ensure JWT secret key is long and secure
+            String secretKey = params.getString(SECRET_KEY);
+            if (!PasswordUtils.isStrongPassword(secretKey, 30)) {
```
Comment (j-coll) [inline opencga-catalog/src/main/java/org/opencb/opencga/catalog/db/mongodb/MetaMongoDBAdaptor.java:235]: you should use here the SECRET_KEY_MIN_LENGTH

File: opencga-catalog/src/main/java/org/opencb/opencga/catalog/exceptions/CatalogDBException.java [hunk 2]
```diff
@@ -133,4 +133,9 @@ public static CatalogDBException cannotUpdateMultipleEntries(String field, Strin
         return new CatalogDBException("Update " + entry + ": Cannot update '" + field + "' parameter for multiple entries");
     }
 
+    public static CatalogDBException jwtSecretKeyException() {
+        return new CatalogDBException("JWT secret key should be at least 30 characters long and contain at least 1 upper case, 1 lower "
```
Comment (j-coll) [inline opencga-catalog/src/main/java/org/opencb/opencga/catalog/exceptions/CatalogDBException.java:137]: you should use here the SECRET_KEY_MIN_LENGTH

File: opencga-catalog/src/main/java/org/opencb/opencga/catalog/auth/authentication/JwtManager.java [hunk 3]
```diff
@@ -35,7 +35,10 @@ public class JwtManager {
     private Key privateKey;
     private Key publicKey;
 
-    private Logger logger;
+    private final Logger logger;
+
+    // 32 characters to ensure it is at least 256 bytes long
```
Comment (j-coll) [inline opencga-catalog/src/main/java/org/opencb/opencga/catalog/auth/authentication/JwtManager.java:40]: bytes or bits?

**Review by j-coll — CHANGES_REQUESTED** [review]

File: opencga-catalog/src/main/java/org/opencb/opencga/catalog/auth/authentication/JwtManager.java (hunk 3 above)
Comment (pfurio) [reply opencga-catalog/src/main/java/org/opencb/opencga/catalog/auth/authentication/JwtManager.java]: fixed

File: opencga-catalog/src/main/java/org/opencb/opencga/catalog/exceptions/CatalogDBException.java (hunk 2 above)
Comment (pfurio) [reply opencga-catalog/src/main/java/org/opencb/opencga/catalog/exceptions/CatalogDBException.java]: fixed

File: opencga-catalog/src/main/java/org/opencb/opencga/catalog/db/mongodb/MetaMongoDBAdaptor.java (hunk 1 above)
Comment (pfurio) [reply opencga-catalog/src/main/java/org/opencb/opencga/catalog/db/mongodb/MetaMongoDBAdaptor.java]: fixed

**Pull request merged**